### PR TITLE
BS-13249 IE loses the focus of the input fields when a click is done …

### PR DESCRIPTION
**Pull request by ntith**
…on the load image on the screen during loading a Task.

### CODE REVIEW

File: forms/forms-view/src/main/java/org/bonitasoft/forms/client/view/controller/FormPagesViewController.java [hunk 1]
```diff
@@ -216,6 +218,18 @@
     protected Map<String, Object> urlContext;
 
     /**
+     * Widget type to focused when loading is hidden
+     *
+     */
+    protected List<WidgetType> widgetTypesToFocus = Arrays.asList(
+            WidgetType.TEXT,
```
Comment (abirembaut) [inline forms/forms-view/src/main/java/org/bonitasoft/forms/client/view/controller/FormPagesViewController.java]: TEXT is a readonly widget

File: forms/forms-view/src/main/java/org/bonitasoft/forms/client/view/controller/FormPagesViewController.java [hunk 2]
```diff
@@ -216,6 +218,18 @@
...
+            WidgetType.TEXTAREA,
+            WidgetType.TEXTBOX,
+            WidgetType.SUGGESTBOX,
+            WidgetType.RICH_TEXTAREA,
+            WidgetType.SUGGESTBOX_ASYNC);
+
```
Comment (abirembaut) [inline forms/forms-view/src/main/java/org/bonitasoft/forms/client/view/controller/FormPagesViewController.java]: Shouldn't we add DATE, FILEUPLOAD, PASSWORD ?
And are we sure ther is no problem with CHECKBOX, CHECKBOX_GROUP, RADIOBUTTON_GROUP, LISTBOX_SIMPLE, LISTBOX_MULTIPLE ?

**Comment by ntith** [comment]
You are right for password
- For date type, I think we must not focus automatically, because it could open the calendar
- Same comment for file upload, I think it will open the system file picker

Checkbox & radio buttons are ok

I will create a bar to check all fields

File: forms/forms-view/src/main/java/org/bonitasoft/forms/client/view/controller/FormPagesViewController.java [hunk 3]
```diff
@@ -392,9 +406,29 @@ protected void buildPage(final ReducedFormPage formPage, final boolean hasAlread
...
+     * Focus the first field "focusable"
+     */
+    protected void focusFirstField() {
+        final Iterator<Entry<String, FormFieldWidget>> fieldWidgestIterator = fieldWidgets.entrySet().iterator();
+
+        boolean focused = false;
+        while (fieldWidgestIterator.hasNext() && !focused) {
+            final Entry<String, FormFieldWidget> reducedFormWidget = fieldWidgestIterator.next();
+            final FormFieldWidget formFieldWidget = reducedFormWidget.getValue();
+            if (formFieldWidget!=null &&  formFieldWidget.getType()!=null && widgetTypesToFocus.contains(formFieldWidget.getType())) {
```
Comment (abirembaut) [inline forms/forms-view/src/main/java/org/bonitasoft/forms/client/view/controller/FormPagesViewController.java]: - indentation to fix
- I think the second null check is nor necessary (the type can never be null and event if if was, there is no null value in widgetTypesToFocus)

File: portal/src/main/java/org/bonitasoft/web/toolkit/client/ui/component/form/AbstractForm.java [hunk 4]
```diff
@@ -538,23 +536,36 @@ protected Element makeElement() {
...
                 return false;
             }
 
         });
-
+        addFocusEventForIE(form);
         return form;
     }
 
+    public boolean isInternetExplorer() {
```
Comment (abirembaut) [inline portal/src/main/java/org/bonitasoft/web/toolkit/client/ui/component/form/AbstractForm.java]: this method doesn't seem to be used anywhere

**Pull request merged**